pkg/helm_resources_manager/helm_resources_manager.go

```diff
@@ -44,25 +48,69 @@ type helmResourcesManager struct {
 
 var _ HelmResourcesManager = &helmResourcesManager{}
 
-func NewHelmResourcesManager() HelmResourcesManager {
-	return &helmResourcesManager{
-		eventCh:  make(chan ReleaseStatusEvent),
-		monitors: make(map[string]*ResourcesMonitor),
+func NewHelmResourcesManager(ctx context.Context, kclient *klient.Client) (HelmResourcesManager, error) {
+	//nolint:govet
+	cctx, cancel := context.WithCancel(ctx)
+	if kclient == nil {
+		//nolint:govet
+		return nil, fmt.Errorf("kube client not set")
 	}
-}
 
-func (hm *helmResourcesManager) WithKubeClient(client *klient.Client) {
-	hm.kubeClient = client
+	cfg := kclient.RestConfig()
+	defaultLabelSelector, err := labels.Parse(app.ExtraLabels)
+	if err != nil {
+		return nil, err
+	}
+	cache, err := cr_cache.New(cfg, cr_cache.Options{
+		DefaultLabelSelector: defaultLabelSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	go cache.Start(cctx)
+	log.Debug("Helm resource manager: cache's been started")
+	if synced := cache.WaitForCacheSync(cctx); !synced {
+		return nil, fmt.Errorf("Couldn't sync helm resource informer cache")
+	}
+	log.Debug("Helm resourcer manager: cache has been synced")
+
+	return &helmResourcesManager{
+		eventCh:    make(chan ReleaseStatusEvent),
+		monitors:   make(map[string]*ResourcesMonitor),
+		ctx:        cctx,
+		cancel:     cancel,
+		kubeClient: kclient,
+		cache:      cache,
+	}, nil
+}
+
+func (hm *helmResourcesManager) WithCache(ctx context.Context) error {
```

Comment:
This function is not used anywhere. Why do we need it?

Reply:
my bad, I forgot to delete it after merging WithClient and WithCache into NewHelmResourcesManager